src/immutable/state/core/index.ts

```diff
@@ -1,32 +1,35 @@
-import {
-  RecordOf,
-  Collection as CollectionType,
-  Map as ImmutableMapType,
-} from 'immutable';
+import { Record, Map as ImmutableMap } from 'immutable';
 
-import { CoreTransactionsRecord } from './CoreTransactions';
-import { GasPricesRecord } from './GasPrices';
-import { NetworkRecordType } from './Network';
-import { DataRecordType } from '../../Data';
-import { CoreMessagesRecord } from './Messages';
-import { ConnectionRecordType } from './Connection';
+import { Connection, ConnectionRecord } from '../../Connection';
+import { FetchableData, FetchableDataRecord } from '../../FetchableData';
+import { CoreTransactions, CoreTransactionsRecord } from './CoreTransactions';
+import { GasPrices, GasPricesRecord } from './GasPrices';
+import { Network, NetworkRecord } from './Network';
+import { CoreMessages, CoreMessagesRecord } from './Messages';
 
 export * from './GasPrices';
 export * from './CoreTransactions';
 export * from './Network';
 export * from './Messages';
-export * from './Connection';
 
-export type IpfsDataType = ImmutableMapType<string, DataRecordType<string>>;
+export type IpfsDataType = ImmutableMap<string, FetchableDataRecord<string>>;
 
 export type CoreStateProps = {
+  connection: ConnectionRecord;
   gasPrices: GasPricesRecord;
-  transactions: CoreTransactionsRecord;
-  network: DataRecordType<NetworkRecordType>;
   ipfsData: IpfsDataType;
   messages: CoreMessagesRecord;
-  connection: ConnectionRecordType;
+  network: FetchableDataRecord<NetworkRecord>;
+  transactions: CoreTransactionsRecord;
 };
 
-export type CoreStateRecord = CollectionType<any, any> &
-  RecordOf<CoreStateProps>;
+export class CoreStateRecord extends Record<CoreStateProps>({
```

Comment:
I thought `combineReducers` would set these default values, but this is fine too.